src/js/preshuffle.s

```diff
@@ -983,8 +987,12 @@ ExitWithoutDrawingEXP:
 .reloc
 RemoveEnemyAndPlaySFX:
   ;; the last attacked enemy is getting cleared out so we want to update the HP display
-  jsr ClearCurrentEnemyHPSlot
-  lda #SFX_MONSTER_HIT
+  ;; Check first if the current slot == this obj slot (stored in $62 at this point)
+  lda CurrentEnemySlot
```

Comment:
Can you expand in the comment a bit about why this might not be the case?  My understanding is that these should (generally) be the same.  Does it have to do with hitting multiple enemies on the same frame?